firmware/src/vendor.c

```diff
@@ -103,6 +106,16 @@ bool handle_force_fpga_offline(uint8_t rhport, tusb_control_request_t const* req
 }
 
 
+/**
+ * Request Apollo to honor FPGA advertisement messages.
+ */
+bool handle_honor_fpga_adv(uint8_t rhport, tusb_control_request_t const* request)
+{
+	honor_fpga_adv();
```

Comment:
Calling `honor_fpga_adv` here can make Apollo switch off its own USB connection before it has finished responding to the vendor request. This call needs to be moved to `handle_vendor_request_complete`.

Reply:
Fixed in 3fae13bf98a5c40e65d9e54dd997c94933624dfd.